[Automation] native-image run fails for com.github.ben-manes.caffeine:caffeine:2.9.3

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.9.3" ./gradlew clean test -Pcoordinates="com.github.ben-manes.caffeine:caffeine:2.9.3"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25727047580/job/75542581098
Last 300 lines of the log:
```console
    ]
  }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportAccessedField(MissingReflectionRegistrationUtils.java:79)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.UnsafeFieldUtil.getFieldOffset(UnsafeFieldUtil.java:46)
       java.base@25.0.3/jdk.internal.misc.Unsafe.staticFieldOffset(Unsafe.java:49)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.fieldaccessor.UnsafeFieldAccessorImpl.<init>(UnsafeFieldAccessorImpl.java:56)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.fieldaccessor.UnsafeStaticFieldAccessorImpl.<init>(UnsafeStaticFieldAccessorImpl.java:42)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.fieldaccessor.UnsafeQualifiedStaticFieldAccessorImpl.<init>(UnsafeQualifiedStaticFieldAccessorImpl.java:39)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.fieldaccessor.UnsafeQualifiedStaticObjectFieldAccessorImpl.<init>(UnsafeQualifiedStaticObjectFieldAccessorImpl.java:33)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.fieldaccessor.UnsafeFieldAccessorFactory.newFieldAccessor(UnsafeFieldAccessorFactory.java:84)
       java.base@25.0.3/jdk.internal.reflect.ReflectionFactory.newFieldAccessor(ReflectionFactory.java:2434)
       java.base@25.0.3/java.lang.reflect.Field.acquireOverrideFieldAccessor(Field.java:1195)
       [...]
  JUnit Jupiter:CaffeineTest:testSpecification()
    MethodSource [className = 'com_github_ben_manes_caffeine.caffeine.CaffeineTest', methodName = 'testSpecification', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.github.benmanes.caffeine.cache.SS
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1717)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1654)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1641)
       com.github.benmanes.caffeine.cache.LocalCacheFactory.newBoundedLocalCache(LocalCacheFactory.java:87)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3423)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalLoadingCache.<init>(BoundedLocalCache.java:3758)
       com.github.benmanes.caffeine.cache.Caffeine.build(Caffeine.java:1104)
       [...]
  JUnit Jupiter:CaffeineTest:testStatistics()
    MethodSource [className = 'com_github_ben_manes_caffeine.caffeine.CaffeineTest', methodName = 'testStatistics', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.github.benmanes.caffeine.cache.SS
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1717)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1654)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1641)
       com.github.benmanes.caffeine.cache.LocalCacheFactory.newBoundedLocalCache(LocalCacheFactory.java:87)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3423)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3419)
       com.github.benmanes.caffeine.cache.Caffeine.build(Caffeine.java:1078)
       [...]
  JUnit Jupiter:CaffeineTest:testCleanup()
    MethodSource [className = 'com_github_ben_manes_caffeine.caffeine.CaffeineTest', methodName = 'testCleanup', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.github.benmanes.caffeine.cache.SS
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1717)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1654)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1641)
       com.github.benmanes.caffeine.cache.LocalCacheFactory.newBoundedLocalCache(LocalCacheFactory.java:87)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3423)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalLoadingCache.<init>(BoundedLocalCache.java:3758)
       com.github.benmanes.caffeine.cache.Caffeine.build(Caffeine.java:1104)
       com_github_ben_manes_caffeine.caffeine.CaffeineTest.testCleanup(CaffeineTest.java:74)
       [...]
  JUnit Jupiter:CaffeineTest:testRefresh()
    MethodSource [className = 'com_github_ben_manes_caffeine.caffeine.CaffeineTest', methodName = 'testRefresh', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.github.benmanes.caffeine.cache.SS
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1717)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1654)
May 12, 2026 10:08:51 AM com.github.benmanes.caffeine.cache.LocalAsyncLoadingCache canBulkLoad
WARNING: Cannot determine if CacheLoader can bulk load
java.lang.NoSuchMethodException: com_github_ben_manes_caffeine.caffeine.PopulationTest$$Lambda/0xe8a92e1b613170f9ba4bfd00b7ded30e0.asyncLoadAll(java.lang.Iterable, java.util.concurrent.Executor)
	at java.base@25.0.3/java.lang.Class.checkMethod(DynamicHub.java:1343)
	at java.base@25.0.3/java.lang.Class.getMethod(DynamicHub.java:1337)
	at com.github.benmanes.caffeine.cache.LocalAsyncLoadingCache.canBulkLoad(LocalAsyncLoadingCache.java:68)
	at com.github.benmanes.caffeine.cache.LocalAsyncLoadingCache.<init>(LocalAsyncLoadingCache.java:51)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalAsyncLoadingCache.<init>(BoundedLocalCache.java:3873)
	at com.github.benmanes.caffeine.cache.Caffeine.buildAsync(Caffeine.java:1203)
	at com_github_ben_manes_caffeine.caffeine.PopulationTest.testAsynchronouslyLoading(PopulationTest.java:64)
	at java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
	at java.base@25.0.3/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base@25.0.3/java.lang.Thread.runWith(Thread.java:1487)
	at java.base@25.0.3/java.lang.Thread.run(Thread.java:1474)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:832)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:808)

       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1641)
       com.github.benmanes.caffeine.cache.LocalCacheFactory.newBoundedLocalCache(LocalCacheFactory.java:87)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3423)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalLoadingCache.<init>(BoundedLocalCache.java:3758)
       com.github.benmanes.caffeine.cache.Caffeine.build(Caffeine.java:1104)
       com_github_ben_manes_caffeine.caffeine.CaffeineTest.testRefresh(CaffeineTest.java:34)
       [...]
  JUnit Jupiter:EvictionTest:testTimeBased()
    MethodSource [className = 'com_github_ben_manes_caffeine.caffeine.EvictionTest', methodName = 'testTimeBased', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.github.benmanes.caffeine.cache.SS
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1717)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1654)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1641)
       com.github.benmanes.caffeine.cache.LocalCacheFactory.newBoundedLocalCache(LocalCacheFactory.java:87)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3423)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalLoadingCache.<init>(BoundedLocalCache.java:3758)
       com.github.benmanes.caffeine.cache.Caffeine.build(Caffeine.java:1104)
       com_github_ben_manes_caffeine.caffeine.EvictionTest.testTimeBased(EvictionTest.java:33)
       [...]
  JUnit Jupiter:EvictionTest:testReferenceBased()
    MethodSource [className = 'com_github_ben_manes_caffeine.caffeine.EvictionTest', methodName = 'testReferenceBased', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.github.benmanes.caffeine.cache.BoundedLocalCache
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1717)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1654)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1641)
       com.github.benmanes.caffeine.cache.LocalCacheFactory.newBoundedLocalCache(LocalCacheFactory.java:87)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3423)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalLoadingCache.<init>(BoundedLocalCache.java:3758)
       com.github.benmanes.caffeine.cache.Caffeine.build(Caffeine.java:1104)
       com_github_ben_manes_caffeine.caffeine.EvictionTest.testReferenceBased(EvictionTest.java:57)
       [...]
  JUnit Jupiter:EvictionTest:testSizeBased()
    MethodSource [className = 'com_github_ben_manes_caffeine.caffeine.EvictionTest', methodName = 'testSizeBased', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.github.benmanes.caffeine.cache.SSMS
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1717)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1654)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1641)
       com.github.benmanes.caffeine.cache.LocalCacheFactory.newBoundedLocalCache(LocalCacheFactory.java:87)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3423)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalLoadingCache.<init>(BoundedLocalCache.java:3758)
       com.github.benmanes.caffeine.cache.Caffeine.build(Caffeine.java:1104)
       com_github_ben_manes_caffeine.caffeine.EvictionTest.testSizeBased(EvictionTest.java:23)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       [...]
  JUnit Jupiter:PolicyTest:testTimeBased()
    MethodSource [className = 'com_github_ben_manes_caffeine.caffeine.PolicyTest', methodName = 'testTimeBased', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.github.benmanes.caffeine.cache.SSMSW
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1717)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1654)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1641)
       com.github.benmanes.caffeine.cache.LocalCacheFactory.newBoundedLocalCache(LocalCacheFactory.java:87)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3423)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3419)
       com.github.benmanes.caffeine.cache.Caffeine.build(Caffeine.java:1078)
       com_github_ben_manes_caffeine.caffeine.PolicyTest.testTimeBased(PolicyTest.java:29)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       [...]
  JUnit Jupiter:PolicyTest:testSizeBased()
    MethodSource [className = 'com_github_ben_manes_caffeine.caffeine.PolicyTest', methodName = 'testSizeBased', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.github.benmanes.caffeine.cache.SSMSW
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1717)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1654)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1641)
       com.github.benmanes.caffeine.cache.LocalCacheFactory.newBoundedLocalCache(LocalCacheFactory.java:87)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3423)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3419)
       com.github.benmanes.caffeine.cache.Caffeine.build(Caffeine.java:1078)
       com_github_ben_manes_caffeine.caffeine.PolicyTest.testSizeBased(PolicyTest.java:21)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       [...]
  JUnit Jupiter:PopulationTest:testExpireAfterWrite()
    MethodSource [className = 'com_github_ben_manes_caffeine.caffeine.PopulationTest', methodName = 'testExpireAfterWrite', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.github.benmanes.caffeine.cache.SSW
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1717)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1654)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1641)
       com.github.benmanes.caffeine.cache.LocalCacheFactory.newBoundedLocalCache(LocalCacheFactory.java:87)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3423)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3419)
       com.github.benmanes.caffeine.cache.Caffeine.build(Caffeine.java:1078)
       com_github_ben_manes_caffeine.caffeine.PopulationTest.testExpireAfterWrite(PopulationTest.java:42)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       [...]
  JUnit Jupiter:PopulationTest:testAsynchronousManual()
    MethodSource [className = 'com_github_ben_manes_caffeine.caffeine.PopulationTest', methodName = 'testAsynchronousManual', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.github.benmanes.caffeine.cache.SSMSW
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1717)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1654)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1641)
       com.github.benmanes.caffeine.cache.LocalCacheFactory.newBoundedLocalCache(LocalCacheFactory.java:87)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalAsyncCache.<init>(BoundedLocalCache.java:3812)
       com.github.benmanes.caffeine.cache.Caffeine.buildAsync(Caffeine.java:1141)
       com_github_ben_manes_caffeine.caffeine.PopulationTest.testAsynchronousManual(PopulationTest.java:52)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
  JUnit Jupiter:PopulationTest:testAsynchronouslyLoading()
    MethodSource [className = 'com_github_ben_manes_caffeine.caffeine.PopulationTest', methodName = 'testAsynchronouslyLoading', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.github.benmanes.caffeine.cache.SSMSW
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1717)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1654)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1641)
       com.github.benmanes.caffeine.cache.LocalCacheFactory.newBoundedLocalCache(LocalCacheFactory.java:87)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalAsyncLoadingCache.<init>(BoundedLocalCache.java:3876)
       com.github.benmanes.caffeine.cache.Caffeine.buildAsync(Caffeine.java:1203)
       com_github_ben_manes_caffeine.caffeine.PopulationTest.testAsynchronouslyLoading(PopulationTest.java:64)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
  JUnit Jupiter:PopulationTest:testManual()
    MethodSource [className = 'com_github_ben_manes_caffeine.caffeine.PopulationTest', methodName = 'testManual', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.github.benmanes.caffeine.cache.SSMSW
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1717)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1654)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1641)
       com.github.benmanes.caffeine.cache.LocalCacheFactory.newBoundedLocalCache(LocalCacheFactory.java:87)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3423)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3419)
       com.github.benmanes.caffeine.cache.Caffeine.build(Caffeine.java:1078)
       com_github_ben_manes_caffeine.caffeine.PopulationTest.testManual(PopulationTest.java:30)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       [...]
  JUnit Jupiter:RemovalTest:testExplicitRemovals()
    MethodSource [className = 'com_github_ben_manes_caffeine.caffeine.RemovalTest', methodName = 'testExplicitRemovals', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.github.benmanes.caffeine.cache.SSMSW
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1717)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1654)
       java.base@25.0.3/java.lang.Class.forName(DynamicHub.java:1641)
       com.github.benmanes.caffeine.cache.LocalCacheFactory.newBoundedLocalCache(LocalCacheFactory.java:87)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3423)
       com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3419)

       com.github.benmanes.caffeine.cache.Caffeine.build(Caffeine.java:1078)
       com_github_ben_manes_caffeine.caffeine.RemovalTest.testExplicitRemovals(RemovalTest.java:25)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       [...]

Test run finished after 8 ms
[         7 containers found      ]
[         0 containers skipped    ]
[         7 containers started    ]
[         0 containers aborted    ]
[         7 containers successful ]
[         0 containers failed     ]
[        16 tests found           ]
[         0 tests skipped         ]
[        16 tests started         ]
[         0 tests aborted         ]
[         1 tests successful      ]
[        15 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.github.ben-manes.caffeine/caffeine/2.9.3/build/reports/problems/problems-report.html
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.github.ben-manes.caffeine/caffeine/2.9.3/build/native/nativeTestCompile/com.github.ben-manes.caffeine.caffeine_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 25s
17 actionable tasks: 7 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for com.github.ben-manes.caffeine:caffeine:2.9.3 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 27s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.9.3 at 2026-05-12T10:08:51Z with exit code 1 after 88s
❌ ERROR: 'native-image run' for 2.9.3 failed with exit code 1.
FAILED[native-image run][2.9.3][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.9.3" ./gradlew clean test -Pcoordinates="com.github.ben-manes.caffeine:caffeine:2.9.3"]
```

--- a/metadata/com.github.ben-manes.caffeine/caffeine/2.9.3/reachability-metadata.json
+++ b/metadata/com.github.ben-manes.caffeine/caffeine/2.9.3/reachability-metadata.json
@@ -510,6 +510,17 @@
           "name": "threadLocalRandomProbe"
         }
       ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.github.benmanes.caffeine.cache.UnsafeAccess"
+      },
+      "type": "sun.misc.Unsafe",
+      "fields": [
+        {
+          "name": "theUnsafe"
+        }
+      ]
     }
   ]
 }

--- a/metadata/com.github.ben-manes.caffeine/caffeine/index.json
+++ b/metadata/com.github.ben-manes.caffeine/caffeine/index.json
@@ -28,7 +28,7 @@
   {
     "metadata-version" : "2.9.3",
     "source-code-url" : "https://repo1.maven.org/maven2/com/github/ben-manes/caffeine/caffeine/$version$/caffeine-$version$-sources.jar",
-    "repository-url" : "https://github.com/ben-manes/caffeine/tree/v2.9.3",
+    "repository-url" : "https://github.com/ben-manes/caffeine",
     "test-code-url" : "https://repo1.maven.org/maven2/com/github/ben-manes/caffeine/caffeine/$version$/caffeine-$version$-test.jar",
     "documentation-url" : "https://repo1.maven.org/maven2/com/github/ben-manes/caffeine/caffeine/$version$/caffeine-$version$-javadoc.jar",
     "description" : "Caffeine is a high-performance Java caching library that provides an in-memory cache with a Guava-inspired API. It supports automatic loading, size-based eviction, time-based expiration, refresh, reference-based entries, removal notifications, and cache statistics.",

--- a/stats/com.github.ben-manes.caffeine/caffeine/2.9.3/stats.json
+++ b/stats/com.github.ben-manes.caffeine/caffeine/2.9.3/stats.json
@@ -15,21 +15,21 @@
     },
     "libraryCoverage" : {
       "instruction" : {
-        "covered" : 8754,
-        "missed" : 41144,
-        "ratio" : 0.175438,
+        "covered" : 8771,
+        "missed" : 41127,
+        "ratio" : 0.175779,
         "total" : 49898
       },
       "line" : {
-        "covered" : 2038,
-        "missed" : 11039,
-        "ratio" : 0.155846,
+        "covered" : 2047,
+        "missed" : 11030,
+        "ratio" : 0.156534,
         "total" : 13077
       },
       "method" : {
-        "covered" : 650,
-        "missed" : 5093,
-        "ratio" : 0.113181,
+        "covered" : 653,
+        "missed" : 5090,
+        "ratio" : 0.113704,
         "total" : 5743
       }
     }

